[Chloe Seah] iP

chloesyy opened the pull request.

Commit:
Add Level 0: Greet - a simple skeleton of which Duke simply greets and exit

Commit:
Level-1: Add echo and exit features to Duke

Commit:
Complete Level-2. Add the ability to store whatever text entered by the user and display them back to the user when requested.

Add tasks array, change greetings.

Commit:
Complete Level-3. Add the ability to mark tasks as done.

Add Task class. Make tasks array and count static variables to Duke class.

Commit:
Tweak the code to comply with coding standard. Customise TaskThomas' lines and personality.

Commit:
Complete Level-4. Add Todo, Event, Deadline classes which inherit from Task class.

Commit:
Complete A-CodeQuality. Improve the code quality by referring to the code quality and standards learnt.

thesamchris commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,16 @@
+public class Deadline extends Task {
+    private static final String DESCRIPTOR = "[D]";
```
nice job, i like how clean the code is 👍

thesamchris reviewed (commented):
very clean and well organised code that is easy to read ☺️

Commit:
Handle StringIndexOutOfBoundsException when adding tasks.

Commit:
Handle exceptions when adding deadline and events without /by or /at. Also handle exceptions when marking task as done without using task number

Commit:
Handle exceptions where task number is not listed when marking it as done. Also handle exceptions where task number = 0.

Commit:
Merge branch 'branch-Level-5'

Commit:
Put all classes in package duke. Put all task classes and sub-classes in package duke.task.

Change isDone boolean to public so that it can be assessed from Duke class.

Commit:
Merge branch 'branch-A-Packages'

Commit:
Use ArrayList<Task> to store tasks instead of array.

Commit:
Add delete command so that user can delete tasks from list.

Commit:
Save data into separate txt file.

Commit:
Merge branch 'branch-Level-6'

Commit:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/Duke.java

Keep deleteTask method and tasks as ArrayList from Level-6 branch.
Keep file saving methods from Level-7 branch.

Commit:
Add code changes for JAR file.

Commit:
Change [Y] and [N] symbols to ticks and crosses when marking a task as done. Add lines in between user's commands to make it less messy.

Commit:
Add Ui Class to contain all print messages which interacts with the user.

Commit:
Add Storage Class to store all methods which deal with loading tasks from file and saving tasks to file

Commit:
Add more OOP - include Command Class and command package to execute different types of commands.
Also add TaskList Class to contain the taskList and also contain operations to add/delete tasks.

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,314 @@
+
+    /*
+    MAIN METHOD
+     */
```
This comment is a little unnecessary.

jtankw3 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,25 @@
+public abstract class Task {
+    protected String description;
+    public boolean isDone;
+    protected String DESCRIPTOR;
```
You should only use caps when the variable is a constant, ie. has the `final` keyword.

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,314 @@
+
+    /*
+    Loads the file and creates new file if file does not exist
+     */
```
The formatting of your javadoc comments are a little off.

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,314 @@
+            file.createNewFile();
+        }
+
+        Scanner read = new Scanner(file);
```
Generally for variable names we want to use nouns

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,314 @@
+    MAIN METHOD
+     */
+    public static void main(String[] args) {
+        String line;
```
`line` may not be the best name here. Is there a better name you can think of?

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,314 @@
+    public static void deleteTask(String line) {
+        int taskNum = 0;
+        try {
+            taskNum = Integer.parseInt(line.substring(7));
```
`7` is a magic number

jtankw3 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,314 @@
+    }
+
+
+
```
Be careful of having extra empty lines.

jtankw3 commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,24 @@
+
+    @Override
+    public String toString() {
+        return DESCRIPTOR + " | " + (this.isDone ? "1 | " : "0 | ") + this.description;
```
You do not need to use the `this` keyword unless it is ambiguous (shadowed by method parameter)

jtankw3 reviewed (commented):
Great job, your code is very readable and organized. I like how you put in the effort to add comments for most of your public methods. Just take note of the javadoc formatting and naming conventions.

Commit:
Make ammendments to code standards and code quality according to pull request comments.

Commit:
Add LocalDate to store deadline and event dates as dates instead of strings

Commit:
Add listing by date for deadlines and events. User can list tasks for a specific date using the command "list yyyy-mm-dd".

Also fix bug where todo task is added even when command string is empty.

Commit:
Add find function where user can find tasks containing a certain substring.

Commit:
Add JavaDoc comments to most non-private methods and all classes.

chloesyy committed:
Merge pull request #1 from chloesyy/branch-Level-8

Merge branch level 8

Commit:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/task/Task.java

chloesyy committed:
Merge pull request #2 from chloesyy/branch-Level-9

Merge branch level 9

Commit:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/command/AddCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java

chloesyy committed:
Merge pull request #3 from chloesyy/branch-A-JavaDoc

Merge branch A-JavaDoc

chloesyy committed:
Set theme jekyll-theme-dinky

Commit:
Add user guide.

Commit:
Merge branch 'master' of https://github.com/chloesyy/ip

Commit:
Make minor changes to UI print lines. Also make changes accordingly in README file.

chloesyy committed:
Set theme jekyll-theme-merlot

Commit:
Make changes to README file.

Commit:
Merge branch 'master' of https://github.com/chloesyy/ip
